nexus_constructor/StatusConsumer.py

```diff
@@ -0,0 +1,111 @@
+from kafka import KafkaConsumer, TopicPartition
+import threading
+from copy import copy
+import time
+from kafka.errors import NoBrokersAvailable
+import json
+
+
+class StatusConsumer:
+    def __init__(self, broker, topic):
+        self.broker = broker
+        self.topic = topic
+        self.connected = False
+        self.filewriters = {}
+        self.files = {}
+        self.status_lock = threading.Lock()
+        self.run_thread = True
+
+        self.thread = threading.Thread(target=self.status_thread)
+        self.thread.start()
+
+    def __del__(self):
+        self.run_thread = False
+        self.thread.join()
+
+    def isConnected(self):
+        self.status_lock.acquire()
+        return_status = copy(self.connected)
+        self.status_lock.release()
+        return return_status
+
+    def getFilewriters(self):
```

Comment:
could use `@property` here along with `@property.setter`? for each of these functions